android/src/main/java/com/gettipsi/stripe/GoogleApiPayFlowImpl.java

```diff
@@ -0,0 +1,233 @@
+import com.facebook.react.bridge.ReadableMap;
+import com.gettipsi.stripe.util.ArgCheck;
+import com.gettipsi.stripe.util.Converters;
+import com.gettipsi.stripe.util.Fun0;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.Status;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.wallet.AutoResolveHelper;
+import com.google.android.gms.wallet.CardRequirements;
+import com.google.android.gms.wallet.IsReadyToPayRequest;
+import com.google.android.gms.wallet.PaymentData;
+import com.google.android.gms.wallet.PaymentDataRequest;
+import com.google.android.gms.wallet.PaymentMethodTokenizationParameters;
+import com.google.android.gms.wallet.PaymentsClient;
+import com.google.android.gms.wallet.ShippingAddressRequirements;
+import com.google.android.gms.wallet.TransactionInfo;
+import com.google.android.gms.wallet.Wallet;
+import com.google.android.gms.wallet.WalletConstants;
+import com.stripe.android.BuildConfig;
+import com.stripe.android.model.Token;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static com.gettipsi.stripe.util.Converters.convertTokenToWritableMap;
+import static com.gettipsi.stripe.util.Converters.getAllowedShippingCountryCodes;
+import static com.gettipsi.stripe.util.PayParams.CURRENCY_CODE;
+import static com.gettipsi.stripe.util.PayParams.SHIPPING_ADDRESS_REQUIRED;
+import static com.gettipsi.stripe.util.PayParams.TOTAL_PRICE;
+
+/**
+ * Created by ngoriachev on 13/03/2018.
+ * see https://developers.google.com/pay/api/tutorial
+ */
+public final class GoogleApiPayFlowImpl extends PayFlow {
+
+  private static final String TAG = GoogleApiPayFlowImpl.class.getSimpleName();
+  private static final int LOAD_PAYMENT_DATA_REQUEST_CODE = 100500;
+
+  private PaymentsClient mPaymentsClient;
+  private Promise payPromise;
+
+  public GoogleApiPayFlowImpl(@NonNull Fun0<Activity> activityProvider) {
+    super(activityProvider);
+  }
+
+  private PaymentsClient createPaymentsClient(@NonNull Activity activity) {
+    return Wallet.getPaymentsClient(
+      activity,
+      new Wallet.WalletOptions.Builder().setEnvironment(getEnvironment()).build());
+  }
+
+  private void isReadyToPay(@NonNull Activity activity, @NonNull final Promise promise) {
+    ArgCheck.nonNull(activity);
+    ArgCheck.nonNull(promise);
+
+    IsReadyToPayRequest request =
+      IsReadyToPayRequest.newBuilder()
+        .addAllowedPaymentMethod(WalletConstants.PAYMENT_METHOD_CARD)
+        .addAllowedPaymentMethod(WalletConstants.PAYMENT_METHOD_TOKENIZED_CARD)
+        .build();
+    mPaymentsClient = createPaymentsClient(activity);
+    Task<Boolean> task = mPaymentsClient.isReadyToPay(request);
+    task.addOnCompleteListener(
+      new OnCompleteListener<Boolean>() {
+        public void onComplete(Task<Boolean> task) {
+          try {
+            boolean result = task.getResult(ApiException.class);
```

Comment:
Can we just:
`promise.resolve(task.getResult(ApiException.class))`
?

Reply:
@paynd I left it as is for simpler logging insertion later. Please let me know if using cheap stack-var is critical for you.

Reply:
@ngoryachev Not at all. It looks forgotten to simplify.